FirebaseAdmin/FirebaseAdmin.Tests/Auth/Providers/OidcProviderConfigTest.cs

```diff
@@ -0,0 +1,331 @@
+        public async Task CreateConfigInvalidArgs(OidcProviderConfigArgs args, string expected)
+        {
+            var auth = CreateFirebaseAuth();
+
+            var exception = await Assert.ThrowsAsync<ArgumentException>(
+                () => auth.CreateProviderConfigAsync(args));
+            Assert.Equal(expected, exception.Message);
+        }
+
+        [Fact]
+        public async Task CreateConfigUnknownError()
+        {
+            var handler = new MockMessageHandler()
+            {
+                StatusCode = HttpStatusCode.InternalServerError,
+                Response = UnknownErrorResponse,
+            };
+            var auth = CreateFirebaseAuth(handler);
+            var args = new OidcProviderConfigArgs()
+            {
+                ProviderId = "oidc.provider",
+                ClientId = "CLIENT_ID",
+                Issuer = "https://oidc.com/issuer",
+            };
+
+            var exception = await Assert.ThrowsAsync<FirebaseAuthException>(
+                () => auth.CreateProviderConfigAsync(args));
+            Assert.Equal(ErrorCode.Internal, exception.ErrorCode);
+            Assert.Null(exception.AuthErrorCode);
+            Assert.StartsWith(
+                "Unexpected HTTP response with status: 500 (InternalServerError)",
+                exception.Message);
+            Assert.NotNull(exception.HttpResponse);
+            Assert.Null(exception.InnerException);
+        }
+
+        internal static FirebaseAuth CreateFirebaseAuth(HttpMessageHandler handler = null)
+        {
+            var providerConfigManager = new ProviderConfigManager(new ProviderConfigManager.Args
+            {
+                Credential = MockCredential,
+                ProjectId = "project1",
+                ClientFactory = new MockHttpClientFactory(handler ?? new MockMessageHandler()),
+                RetryOptions = RetryOptions.NoBackOff,
+            });
+            var args = FirebaseAuth.Args.CreateDefault();
+            args.ProviderConfigManager = new Lazy<ProviderConfigManager>(providerConfigManager);
+            return new FirebaseAuth(args);
+        }
+
+        private void AssertOidcProviderConfig(OidcProviderConfig provider)
+        {
+            Assert.Equal("oidc.provider", provider.ProviderId);
+            Assert.Equal("oidcProviderName", provider.DisplayName);
+            Assert.True(provider.Enabled);
+            Assert.Equal("CLIENT_ID", provider.ClientId);
+            Assert.Equal("https://oidc.com/issuer", provider.Issuer);
+        }
+
+        private void AssertClientVersionHeader(MockMessageHandler.IncomingRequest request)
+        {
+            Assert.Contains(ClientVersion, request.Headers.GetValues("X-Client-Version"));
+        }
+
+        public class InvalidCreateArgs : IEnumerable<object[]>
+        {
+            public IEnumerator<object[]> GetEnumerator()
+            {
+                // { InvalidInput, ExpectedError }
```

Comment:
Does this comment intend to summarize the errors expected?

Reply:
Updated the comment to clarify the purpose.